Source/missiles.cpp

```diff
@@ -30,6 +30,7 @@ int AvailableMissiles[MAXMISSILES];
 Missile Missiles[MAXMISSILES];
 int ActiveMissileCount;
 bool MissilePreFlag;
+bool collisionFlag;

```

Comment:
Don't use a global variable to hold state for a single missile. If multiple elemental arrows are active, they'll be vying for control over the state of the global variable.

It should be much simpler if we apply the elemental damage to the same target at the point where they take the physical damage, in which case we wouldn't need to use any additional state or spread damage out across multiple frames.

Reply:
The one downside to this though is that these arrows' elemental damage wouldn't be able to bypass a block animation.  From a PvE perspective, this is inconsequential and likely beneficial, however this is a very helpful PvP feature

Reply:
That's not true, actually. There are two ways to bypass the block animation. The first is to use the `shift` flag to bypass the block check, like how Firewall or Acid Puddles do it. The second way is to apply the damage after the player has already started blocking.

In Diablo, if the player blocks the arrow, they won't be able to block the elemental damage regardless of whether it occurs in the same frame. If the player gets hit by the arrow, they will have a chance to block the elemental damage regardless of whether it occurs in the same frame. In either case, the frame in which the damage occurs doesn't matter.

In Hellfire, blocked projectiles don't get deleted so the arrow would pass right through and the elemental damage would never be applied. If the player gets hit by the projectile, they'd have a chance of blocking the elemental damage the same as in Diablo. Once again, the frame in which the damage occurs doesn't matter.

Reply:
Okay, sounds like the way to achieve the desired effect then is to apply elemental damage immediately after the arrow is blocked to allow the elemental damage to still hit?

Reply:
> Don't use a global variable to hold state for a single missile. If multiple elemental arrows are active, they'll be vying for control over the state of the global variable.
> 
> It should be much simpler if we apply the elemental damage to the same target at the point where they take the physical damage, in which case we wouldn't need to use any additional state or spread damage out across multiple frames.

Removed bool

Reply:
I think the collision needs to be rewritten before we can check for collision in the same tick twice, since the second animation isn't moved to the target tile until the next tick.